### PR TITLE
Adding support for storagepool tier

### PR DESCRIPTION
**Pull request by trisha-dell**
# Description
Adding support for storagepool tier

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
powerscale_storagepool_tier

##### OUTPUT
Screenshot 2024-11-19 230112

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 80% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Unit tests
- [X] Acceptance tests

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: examples/resources/powerscale_storagepool_tier/provider.tf [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+terraform {
+  required_providers {
+    powerscale = {
+      source = "registry.terraform.io/dell/powerscale"
+      version = "1.0.0"
+    }
+  }
+}
+
+provider "powerscale" {
```
Comment (shenda1) [inline examples/resources/powerscale_storagepool_tier/provider.tf:26]: Dont expose the IP. You can use the variables instead.

File: examples/resources/powerscale_storagepool_tier/resource.tf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+resource "powerscale_storagepool_tier" "example" {
+  # Required field both for creating and updating
+  children = [
+        "x410_34tb_1.6tb-ssd_64gb"
+    ]
+  name = "Sample_terraform_tier_7"
+  transfer_limit_pct = 40
+  #transfer_limit_state = "disabled"
+
+  # Optional attribute for creating and updating
```
Comment (shenda1) [inline examples/resources/powerscale_storagepool_tier/resource.tf:31]: Once comment is removed for transfer_limit_state, all the attributes will be covered.

File: powerscale/provider/storagepool_tier_resource.go [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+	deleteParam := r.client.PscaleOpenAPIClient.StoragepoolApi.DeleteStoragepoolv16StoragepoolTier(ctx, strconv.FormatInt(data.Id.ValueInt64(), 10))
+	_, err := deleteParam.Execute()
+	if err != nil {
+		errStr := constants.DeleteStoragepoolTierErrorMsg + "with error: "
+		message := helper.GetErrorString(err, errStr)
+		resp.Diagnostics.AddError(
+			"Error deleting storage pool",
+			message,
+		)
+	}
```
Comment (shenda1) [inline powerscale/provider/storagepool_tier_resource.go:246]: Is state getting cleared after deletion?

**Review by shenda1 — COMMENTED** [review]
What are the change in the go-client? Dont see any changes in the OpenAPI specs?

**Comment by shenda1** [comment]
@trisha-dell Please upload the coverage report.

File: examples/resources/powerscale_storagepool_tier/resource.tf (hunk 2 above)
Comment (trisha-dell) [reply examples/resources/powerscale_storagepool_tier/resource.tf:31]: transfer_limit_pct and transfer_limit_state are mutually exclusive parameters.

**Comment by AnikaAgiwal2711** [comment]
you'll also have to run make generate

File: powerscale/provider/storagepool_tier_resource.go (hunk 3 above)
Comment (trisha-dell) [reply powerscale/provider/storagepool_tier_resource.go:246]: yes

File: examples/resources/powerscale_storagepool_tier/provider.tf (hunk 1 above)
Comment (trisha-dell) [reply examples/resources/powerscale_storagepool_tier/provider.tf:25]: Done.

**Comment by trisha-dell** [comment]
Acceptance test execution:
Screenshot 2024-11-20 232828

Acceptance test coverage:
Screenshot 2024-11-20 233026

**Comment by shenda1** [comment]
run e2e test StoragepoolTierResource

**Comment by shenda1** [comment]
e2e pass

**Pull request closed without merging**